code/game/objects/items/trash_material.dm

```diff
@@ -0,0 +1,72 @@
+/obj/item/trash/material
+	icon = 'icons/obj/material_trash.dmi'
+	matter = list()
+	var/matter_chances = list()	//List of lists: list(mat_name, chance, amount)
+
+
+/obj/item/trash/material/Initialize()
+	. = ..()
+	if(!matter)
+		matter = list()
+
+	for(var/list/L in matter_chances)
+		if(prob(L[2]))
+			matter |= L[1]
+			matter[L[1]] += max(0, L[3] + rand(-2,2))
+
+
+
+
+/obj/item/trash/material/metal
+	name = "scrap metal"
+	desc = "A piece of metal that can be recycled in an autolathe."
+	icon_state = "metal0"
+	matter_chances = list(
+		list(MAT_STEEL, 100, 15),
+		list(MAT_STEEL, 50, 10),
+		list(MAT_STEEL, 10, 20),
+		list(MAT_PLASTEEL, 10, 5),
+		list(MAT_PLASTEEL, 5, 10)
+	)
+
+/obj/item/trash/material/metal/Initialize()
+	. = ..()
+	icon_state = "metal[rand(4)]"
+
+
+/obj/item/trash/material/circuit
+	name = "burnt circuit"
+	desc = "A burnt circuit that can be recycled in an autolathe."
+	w_class = ITEMSIZE_SMALL
+	icon_state = "circuit0"
+	matter_chances = list(
+		list(MAT_GLASS, 100, 4),
+		list(MAT_GLASS, 50, 3),
+		list(MAT_PLASTIC, 40, 3),
+		list(MAT_SILVER, 18, 3),
+		list(MAT_GOLD, 17, 3),
+		list(MAT_DIAMOND, 4, 2),
+	)
+
+/obj/item/trash/material/circuit/Initialize()
+	. = ..()
+	icon_state = "circuit[rand(3)]"
+
+
+/obj/item/trash/material/device
+	name = "broken device"
+	desc = "A broken device that can be recycled in an autolathe."
```

Comment:
Not sure if this was meant to say protolathe, or if autolathes on virgo can use materaisl beyond steel and glass, but ours can't.

Reply:
Oh, was just copypaste from a Bay-based server. Maybe theirs can. Hmm.

Reply:
I believe protolathes themselves can't recycle any material here either. Its autolathe-only feature and its steel and glass only. Attempting to recycle something that doesn't have steel or glass will just fail, and attempting to recycle something that has more material than just steel or glass, will have that extra material lost.

Reply:
Well, it'll just work for the steel and glass parts then until someone upgrades the autolathes. Should be transparent to players in-game since as far as I'm aware there's no way to tell what materials are coded onto an item. Consider it motivation for someone to upgrade autolathe/protolathe code.

Reply:
Autolathes were going to be the target of my material-expansion eventually, so this just pushes it forward.